config/crd/standard/gateway.networking.k8s.io_gateways.yaml

```diff
@@ -1172,12 +1178,18 @@ spec:
             description: Status defines the current state of Gateway.
             properties:
               addresses:
-                description: Addresses lists the IP addresses that have actually been
-                  bound to the Gateway. These addresses may differ from the addresses
+                description: "Addresses lists the IP addresses that have actually
+                  been bound to the Gateway. These addresses may differ from the addresses
                   in the Spec, e.g. if the Gateway automatically assigns an address
-                  from a reserved pool.
+                  from a reserved pool. \n Implementations that support GatewayRoutability
+                  MUST include an address that has the same routable semantics as
+                  defined in the Gateway spec. \n Implementations MAY add additional
+                  addresses in status, but they MUST be semantically less than the
+                  scope of the requested scope. For example if a user requests a `Private`
+                  routable Gateway then an additional address MAY have a routability
+                  of `Cluster` but MUST NOT include `Private`."
```

Comment:
Not a blocker for this PR, but unfortunate that we don't have a way to limit spec changes like this to experimental CRDs yet: https://github.com/kubernetes-sigs/gateway-api/issues/2160.

Reply:
Related: https://github.com/kubernetes-sigs/controller-tools/issues/705

We could prune fields based on markers

Reply:
Though controller-tools folks don't want to maintain such a feature - so you could do some post-processing of the CRDs

Reply:
Yep, we are already doing some custom generation here (see #2160) we just need to extend it further to support this use case.